ion-schema/src/isl/isl_type_reference.rs

```diff
@@ -247,6 +247,43 @@ impl IslTypeRefImpl {
         }
     }
 
+    fn occurs_from_ion_element(value: &Element, isl_version: IslVersion) -> IonSchemaResult<Range> {
```

Comment:
I'd suggest that this function is better suited as a part of `IslVariablyOccurringTypeRef`.